packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -2,62 +2,395 @@
+### Shapes
+
+- `square` (default)
+- `circular`
+- `rounded`
+
+Border radius for rounded shape can be customized via CSS.
+
+### Size
+
+- `extraSmall`: 20px
+- `small`: 24px
+- `medium` (default): 28px
+- `large`: 32px
+
+Custom size can be set by overriding `width` and `height` of the ColorSwatch or ImageSwatch.
+
+### Spacing
+
+- `small`: 2px
+- `medium` (default): 4px
+
+### States
+
+- `rest`
+- `hover`
+- `pressed`
+- `selected`
+- `selected pressed`
+- `selected hover`
+- `focused`
+- `disabled` - should be used with caution. In case if there are too many disabled swatches it's better to hide them.
+- `empty`
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### SwatchPicker
+
+| Property          | Values                                   | Default   | Purpose                                                      |
+| ----------------- | ---------------------------------------- | --------- | ------------------------------------------------------------ |
+| disabled          | boolean                                  | false     | Whether SwatchPicker is disabled                             |
+| layout            | `row`, `grid`                            | `row`     | Sets layout of the SwatchPicker                              |
+| onColorPreview    | function                                 | undefined | Callback called when color is previewed (on hover or focus). |
+| onColorSelect     | function                                 | undefined | Callback called when color is selected                       |
+| previewId         | `number`, `string`                       |           | Id of previewColor                                           |
+| selectedId        | `number`, `string`                       |           | Id of selected color                                         |
+| shape             | `square`, `circular`, `rounded`          | `square`  | Sets shape                                                   |
+| size              | `extraSmall`, `small`, `medium`, `large` | `medium`  | Defines size of the Swatch cell                              |
+| spacing           | `small`, `medium`                        | `medium`  | Sets spacing between rows and cells                          |
+| defaultSelectedId | `number`, `string`                       |           | Default selected swatch                                      |
+
+| Slots | Values | Default | Description                  |
+| ----- | ------ | ------- | ---------------------------- |
+| root  | `div`  | `div`   | The root of the SwatchPicker |
+
+### ColorSwatch
+
+| Property | Values                                   | Default  | Purpose                          |
+| -------- | ---------------------------------------- | -------- | -------------------------------- |
+| id       | `number`, `string`                       |          | Sets ID of the swatch            |
```

Comment:
why is `SwatchPicker` missing this property ?

Reply:
Not sure if SwatchPicker needs it. `selectedColor` is part of a context. Maybe for ImageSwatch I'll need `selectedId` but it, also, will be part of a context. I thought about IDs for images because swatch might be a small image which will be mapped to a large one